tests/pkg_edhoc_c/initiator.c

```diff
@@ -98,13 +98,19 @@ static int _parse_ipv6_addr(char *addr_str, ipv6_addr_t *addr, uint16_t *netif)
 static ssize_t _send(coap_pkt_t *pkt, size_t len, ipv6_addr_t *addr, uint16_t netif, uint16_t port)
 {
     sock_udp_ep_t remote;
+    sock_udp_t sock;
+    ssize_t res;
 
     remote.family = AF_INET6;
     remote.port = port;
     remote.netif = netif;
     memcpy(&remote.addr.ipv6[0], addr->u8, sizeof(addr->u8));
 
-    return nanocoap_request(pkt, NULL, &remote, len);
+    nanocoap_connect(&sock, NULL, &remote);
```

Comment:
this is what I mean with "more work at the call site". one line became six.

Reply:
Hm but since `nanocoap_get()` is just a wrapper around `nanocoap_request()`, we'll also need two versions of that.

Reply:
Maybe it'd make sense to have sock alternatives backing some of the calls, but not even public, so that the APIs stay endpoint oriented, but the block-wise operation, rather than going through public parts, uses the sock workhorse function of request?